### PR TITLE
Test non-matching joins

### PR DESCRIPTION
**Pull request by alyst**
These tests should fail until JuliaData/CategoricalArrays.jl#111 and #1316 are merged.

### CODE REVIEW

File: test/join.jl [hunk 1]
```diff
@@ -465,4 +465,58 @@ module TestJoin
...
+        @test eltypes(join(l, r, on=:id, kind=:right)) == [Int, Union{CS, Missing}, CS]
+
+        @test join(l, r, on=:id, kind=:outer) ≅
+            DataFrame(id=vcat(l[:id], r[:id]),
+                      cid=vcat(l[:cid], fill(missing, nr)),
+                      cid_1=vcat(fill(missing, nl), r[:cid]))
+        @test eltypes(join(l, r, on=:id, kind=:outer)) == [Int, Union{CS, Missing}, Union{CS, Missing}]
+
+        # joins by cid
+        @test join(l, r, on=:cid, kind=:inner) ≅ DataFrame(id=Int[], cid=similar(l[:cid], 0), id_1=similar(r[:cid], 0))
```
Comment (nalimilan) [inline test/join.jl]: Break line at 92 chars. Same elsewhere.

File: test/join.jl [hunk 2]
```diff
@@ -465,4 +465,58 @@ module TestJoin
         @test join(left, right, on = [:id => :ID, :sid => :SID], kind=:anti) ==
             DataFrame(id = 1:2, sid = string.(1:2))
     end
+
+    @testset "joins with no matching rows" begin
```
Comment (nalimilan) [inline test/join.jl]: "and `CategoricalArray` columns"?

**Comment by nalimilan** [comment]
Should be possible to rebase & merge now.

**Comment by alyst** [comment]
Thanks! Rebased

**Comment by alyst** [comment]
Tests pass on v0.6, nighlies fail early in joins, so it's hard to say whether anything in the new tests needs fixing.

**Pull request merged**